### PR TITLE
Zay1939 feature/add history tab my profile

### PR DESCRIPTION
**Pull request by nikitaKrpch**
# What I did
- Added a new **History** tab to the MyProfile screen allowing users to see all completed listings.
- Unified the top tab bar into a reusable `SelectionRow` with 4 tabs: Info, Listings, Ratings, History.
- Added the new `ProfileHistory` composable to display completed listings.
- Updated UI tests to cover History tab visibility, navigation, and content rendering (AI generated).

# How I did it
- Extended the `ProfileTab` enum with a new `HISTORY` entry.
- Updated `SelectionRow` to compute dynamic text widths and animate the indicator position.
- Modified the main `MyProfileScreen` to include History in the `when(selectedTab)` switch.
- Implemented `ProfileHistory` as a sibling of Listings & Ratings:
  - Filters listings by `!isActive`
  - Displays cards using existing `ProposalCard` / `RequestCard`
  - Handles loading/error/empty states
- AI generated new Compose UI tests
- Updated existing tests to support the 4-tab layout.

# How to verify it
1. Open My Profile screen with a user that has both active and completed listings.
2. Tap the **History** tab.
3. Confirm:
   - The indicator moves under "History" and centers properly.
   - You see "Your History" at the top.
   - Only non-active (completed) listings appear.
4. Switch between Info → Listings → Ratings → History:
   - Indicator animates smoothly.
   - The correct content loads each time.
5. Test edge case:
   - No completed listings: should show “You don’t have any completed listings yet.”

# Demo video
https://github.com/user-attachments/assets/dd5654f7-f637-490c-8ccb-522fad333621

# Pre-merge checklist
The changes I introduced:
- [x] work correctly
- [x] do not break other functionalities
- [x] work correctly on Android
- [x] are fully tested (or have tests added)

### CODE REVIEW

File: app/src/main/java/com/android/sample/ui/profile/MyProfileScreen.kt [hunk 1]
```diff
@@ -503,6 +508,60 @@ private fun ProfileListings(ui: MyProfileUIState, onListingClick: (String) -> Un
...
 }
 
+/**
+ * History section showing the user's completed listings.
+ *
+ * @param ui Current UI state providing listings and profile data for the creator.
+ * @param onListingClick Callback when a listing card is clicked.
+ */
+@Composable
+private fun ProfileHistory(ui: MyProfileUIState, onListingClick: (String) -> Unit) {
```
[the remark this answers is not in the document]
Comment (Zay1939) [reply app/src/main/java/com/android/sample/ui/profile/MyProfileScreen.kt]: Thanks !
